source/grammar/openpulse1.g4

```diff
@@ -0,0 +1,74 @@
+/**** ANTLRv4  grammar for OpenPulse1.0.
+
+The following grammar specifies valid OpenPulse statements within the .*? of
+OpenQASM3 below.
+
+calibrationDefinition
+    : 'defcal' calibrationGrammar? Identifier
+    ( LPAREN calibrationArgumentList? RPAREN )? identifierList
+    returnSignature? LBRACE .*? RBRACE  // for now, match anything inside body
+    ;
+
+specifically, `.*? -> body` below.
+
+****/
+
+grammar openpulse1;
+
+body
+    : (statement)+
+    ;
+
+statement
+    : classicalAssignment
+    | classicalDeclaration
+    | constantDeclaration
+    | aliasStatement
+    | branchingStatement
+    | instruction
+    | pragma
+    ;
+
+instruction
+    : 'play' LPAREN pulse RPAREN SEMICOLON
+    | frameUpdateInstruction
+    | captureInstruction
+    ;
+
+pulse
+    : LBRACKET (Complex COMMA)* Complex RBRACKET
+    ;
+
+frameUpdateInstruction
+    : frame DOT frameAttribute frameAssignmentOperator expression SEMICOLON
```

Comment:
I think the `frameAssignmentOperator` could more generally be any `binaryOperator`. It might make sense, instead of using this, to add `frameExpression : frame DOT frameAttribute` to `expression`. Since these properties are numeric values they could be used in any expression. For instance, you could add a boolean statement compare frequencies or phases.